src/services/legal-services.ts

```diff
@@ -11,11 +11,20 @@ export default class LegalServices {
   /**
    * Fetches business info.
    * @param businessId the business identifier (aka entity inc no)
-   * @returns the axios response
+   * @returns the business info object
    */
-  static async fetchBusinessInfo (businessId: string): Promise<AxiosResponse> {
+  static async fetchBusinessInfo (businessId: string): Promise<BusinessIF> {
     const url = `businesses/${businessId}`
     return axios.get(url)
+      .then(response => {
+        const businessInfo = response?.data?.business as BusinessIF
+        if (!businessInfo) {
+          // eslint-disable-next-line no-console
+          console.log('fetchBusinessInfo() error - invalid response =', response)
+          throw new Error('Invalid business info')
+        }
+        return businessInfo
+      })
```

Comment:
Do we need to return a new Promise here?  If "businessInfo" is not returned from the API, does `fetchBusinessInfo` return a Promise.resolve?

Reply:
Because we're returning the then block of an async function, this is inherently a promise.

Then -> resolved

Catch -> rejected

This code works perfectly and as you expect it to, and it's fairly simple-looking 👍

Reply:
Technically, my comment on line 14 is incorrect -- in fact, what is returned is a promise to return the data object -- but that's pretty wordy so I wrote it this way.

Reply:
If the Axios call is successful but returns bad data, I'm pretty sure Axios returns a Promise.resolve.  In OpenShift (and other hosting platforms), I've seen API calls return successful responses but bad data.  When bad data is returned, we need to return a Promise.reject to get the downstream caller to respond properly.  Didn't you and I test this previously and determine that we needed to `return new Promise ...`?

Reply:
Axios returns an exception for a 4xx or 5xx response, otherwise it returns a resolved promise (then block).

Reply:
What we previously tested is whether the action needs to return a wrapped promise.